pandas/core/indexes/extension.py

```diff
@@ -242,6 +243,10 @@ def repeat(self, repeats, axis=None):
         result = self._data.repeat(repeats, axis=axis)
         return self._shallow_copy(result)
 
+    def insert(self, loc: int, item):
+        # ExtensionIndex subclasses must override Index.insert
```

Comment:
Any reason not to make an abstractmethod?

Reply:
good idea, updated

Reply:
So this I believe is ideal but any concerns on backward compat? Otherwise lgtm

Reply:
> any concerns on backward compat?

Nothing comes to mind

Reply:
Gothca. This is non-public right?

Reply:
`ExtensionIndex` needs to inherit from `abc.ABC` in order for the abstractmethod decorator to actually work, right?
```python
In [1]: import abc

In [2]: class Foo: 
   ...:     @abc.abstractmethod 
   ...:     def some_meth(self): 
   ...:         pass 
   ...: 

In [3]: class Bar(Foo): 
   ...:     pass 
   ...: 

In [4]: bar = Bar()

In [5]: class Foo2(abc.ABC): 
   ...:     @abc.abstractmethod 
   ...:     def some_meth(self): 
   ...:         pass 
   ...: 

In [6]: class Bar2(Foo2): 
   ...:     pass 
   ...:

In [7]: bar2 = Bar2()
---------------------------------------------------------------------------
TypeError: Can't instantiate abstract class Bar2 with abstract methods some_meth
```